packages/hardhat-plugin/src/index.ts

```diff
@@ -227,55 +229,108 @@ task("visualize")
     }
   );
 
-task("ignition-info")
-  .addParam("deploymentId")
-  .addFlag("json", "format as json")
+task("status")
+  .addParam("id")
   .setDescription("Lists the deployed contract addresses of a deployment")
-  .setAction(
-    async (
-      {
-        deploymentId,
-        json: formatAsJson,
-      }: { deploymentId: string; json: boolean },
-      hre
-    ) => {
-      const deploymentDir = path.join(
-        hre.config.paths.ignition,
-        "deployments",
-        deploymentId
+  .setAction(async ({ id }: { id: string }, hre) => {
+    const { status } = await import("@nomicfoundation/ignition-core");
+
+    const deploymentDir = path.join(
+      hre.config.paths.ignition,
+      "deployments",
+      id
+    );
+
+    const statusResult = await status(deploymentDir);
+
+    if (statusResult.started.length > 0) {
+      console.log("");
+      console.log(
+        chalk.bold(`⛔ Deployment ${id} exited early and needs to be rerun`)
       );
-      const deployedAddressesPath = path.join(
-        deploymentDir,
-        "deployed_addresses.json"
+      console.log("");
+
+      console.log(
+        chalk.yellow(
+          "There are futures that were started but have not completed:"
+        )
       );
+      console.log("");
 
-      if (!existsSync(deploymentDir)) {
-        console.error(`No deployment found with id ${deploymentDir}`);
-        process.exit(1);
+      for (const futureId of Object.values(statusResult.started)) {
+        console.log(` - ${futureId}`);
       }
 
-      if (!existsSync(deployedAddressesPath)) {
-        console.log(`No contracts deployed`);
-        process.exit(0);
+      console.log("");
+      return;
+    }
+
+    if (
+      statusResult.timedOut.length > 0 ||
+      statusResult.failed.length > 0 ||
+      statusResult.held.length > 0
+    ) {
+      console.log("");
+      console.log(
+        chalk.bold(`⛔ Deployment ${id} has failed or halted futures`)
```

Comment:
I think using the word halted here can be confusing, as it's the only place we use it.

Reply:
That is a good point. How about "Deployment XXX has either failed or timed out". Or do you think we do a case statement on the text depending on the combination of failed/timed out/held?

Reply:
I added a case statement.